### PR TITLE
ZIP 320 (TEX address) support

### PR DESCRIPTION
**Pull request by str4d**
> **Note**
>This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [ ] Update **documentation** as appropriate (e.g [README.md](../blob/main/README.md), [Architecture.md](../blob/main/docs/Architecture.md), etc.)
- [x] **Run the demo app** and try the changes
- [ ] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

# Reviewer

- [ ] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [ ] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [ ] Review the **documentation**, [README.md](../blob/main/README.md), [Architecture.md](/blob/main/docs/Architecture.md), etc. as appropriate
- [ ] **Run the demo app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the SDK, some aspects require manual testing. If you had to manually test 
something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the demo app to look for build failures or crashes, humans running the demo app are 
more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK

**Comment by str4d** [comment]
[Force-pushed](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/compare/6ae556688970640ff603f327e90b88fb47f0c93b..2a7bba32d9ab2dc426a9ba086e5c1962b2395e24) to fix lints.

**Review by daira — APPROVED** [review]
utACK 2a7bba32d9ab2dc426a9ba086e5c1962b2395e24

**Review by Milan-Cerovsky — APPROVED** [review]
Looks good to me 👍 could u please also update changelog?

File: backend-lib/Cargo.toml [hunk 1]
```diff
@@ -63,10 +63,10 @@ crate-type = ["staticlib", "cdylib"]
...
 shardtree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "337f59179eda51261e9ddfc6b18e8fb84ea277c9" }
-zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_client_backend = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_client_sqlite = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_encoding = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_primitives = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_proofs = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
-zcash_protocol = { git = "https://github.com/zcash/librustzcash.git", rev = "34cf6d286a201caf7dfee29fbee867d547afc486" }
+zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "eaa43b4e2597786317e59863e02bf0701152f398" }
```
Comment (str4d) [inline backend-lib/Cargo.toml:66]: There are necessary bugfixes in the upstream Rust crates, but given that #1533 also alters these revisions, I propose that we merge this PR and #1533, and then I'll make another PR that updates the feature branch with bugfixes.

**Review by str4d — COMMENTED** [review]
ACK for non-me changes.

**Pull request merged**